madgraph/interface/reweight_interface.py

```diff
@@ -1024,58 +1313,258 @@ def change_kinematics(self, event):
+            # LO reweighting    
+            w_orig = self.calculate_matrix_element(event, 0)
+            # reshuffle event for mass effect # external mass only
+            # carefull that new_event can sometimes be = to event 
+            # (i.e. change can be in place)
+            jac, new_event = self.change_kinematics(event)
         
         
-        if event.wgt != 0: # impossible reshuffling
-            w_new =  self.calculate_matrix_element(new_event, 1)
-        else:
-            w_new = 0
+            if event.wgt != 0: # impossible reshuffling
+                w_new =  self.calculate_matrix_element(new_event, 1)
+            else:
+                w_new = 0
+
+            if w_orig == 0:
+                tag, order = event.get_tag_and_order()
+                orig_order, Pdir, hel_dict = self.id_to_path[tag]
+                misc.sprint(w_orig, w_new)
+                misc.sprint(event)
+                misc.sprint(self.invert_momenta(event.get_momenta(orig_order)))
+                misc.sprint(event.get_momenta(orig_order))
+                misc.sprint(event.aqcd)
+                hel_order = event.get_helicity(orig_order)
+                if self.helicity_reweighting and 9 not in hel_order:
+                    nhel = hel_dict[tuple(hel_order)]
+                else:
+                    nhel = 0
+                misc.sprint(nhel, Pdir, hel_dict)                        
+                raise Exception("Invalid matrix element for original computation (weight=0)")
 
-        if w_orig == 0:
-            tag, order = event.get_tag_and_order()
-            orig_order, Pdir, hel_dict = self.id_to_path[tag]
-            misc.sprint(w_orig, w_new)
-            misc.sprint(event)
-            misc.sprint(self.invert_momenta(event.get_momenta(orig_order)))
-            misc.sprint(event.get_momenta(orig_order))
-            misc.sprint(event.aqcd)
-            hel_order = event.get_helicity(orig_order)
-            if self.helicity_reweighting and 9 not in hel_order:
-                nhel = hel_dict[tuple(hel_order)]
+            return {'orig': orig_wgt, '': w_new/w_orig*orig_wgt*jac}
+        else:
+            import copy
+            buff_event=copy.deepcopy(event)
+            orig_wgt = event.wgt
+            w_orig= event.wgt
+            pi=3.141592653589
+            mW = 80.3
+
+            mgcmd = self.mg5cmd
+            import importlib
+            import numpy as np
+
+            for proc in [line[9:].strip() for line in self.banner.proc_card
+                                if line.startswith('generate')]:
+                process, order, final = re.split('\[\s*(.*)\s*\]', proc)
+                n_init = len(re.split('\s+',re.split('>', process)[0])) -1
+                n_final = len(re.split('\s+',re.split('>', process)[1])) -2
+            nexternal = n_init + n_final
+
+            # Remove all propagator particles from the event to be passed to Sud module
+            for ip,part in enumerate(buff_event):
```

Comment:
```suggestion
            for ip,part in enumerate(list(buff_event)):
```
(check that list is ok here)

Your previous code sounds dangerous since you edit a "list" while editing it.
Such behavior is not guaranteed and you might face issue. (including some propagator not removed and out of bound/...)